keras_nlp/tokenizers/byte_pair_tokenizer.py

```diff
@@ -0,0 +1,392 @@
+# Copyright 2022 The KerasNLP Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import json
+from typing import Iterable
+from typing import List
+
+import tensorflow as tf
+import tensorflow_text as tf_text
+
+from keras_nlp.tokenizers import tokenizer
+
+
+def bytes_to_unicode():
+    bs = (
+        list(range(ord("!"), ord("~") + 1))
+        + list(range(ord("¡"), ord("¬") + 1))
+        + list(range(ord("®"), ord("ÿ") + 1))
+    )
+    cs = bs[:]
+    n = 0
+    # removes mapping an int to a whitespace character
+    for b in range(2**8):
+        if b not in bs:
+            bs.append(b)
+            cs.append(2**8 + n)
+            n += 1
+    cs = [chr(n) for n in cs]
+    return bs, cs  # int to string mapping
+
+
+class BytePairTokenizerCache:
```

Comment:
It feels weird to have a cache with no size limit. That's not actually a cahce.

Would this become a complete memory hog on a sufficiently large vocabulary? Does that come up in practice?

I think the lru_cache gpt2 uses does have a max size by default.

Reply:
The lru_cache, at least in this [openai implementation](https://github.com/openai/gpt-2/blob/master/src/encoder.py), is only for the byte2unicode mapping, which would be a fixed size. The python dictionary for the cache is unbounded. We could add a limit to the cache if that is better, but to have an lru_cache would require reimplementing something like a MutableHashTable or DenseHashTable.

Reply:
Ohh got it, thanks! Yeah if the original implementation also has an unbounded cache, that makes me more comfortable.